dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -164,8 +164,8 @@
 	</PropertyGroup>
 
 	<PropertyGroup>
-		<_GlobalizationDataFile Condition="'$(_PlatformName)' != 'macOS' And '$(InvariantGlobalization)' != 'true' And '$(HybridGlobalization)' != 'true' And '$(_GlobalizationDataFile)' == ''">icudt.dat</_GlobalizationDataFile>
-		<_GlobalizationDataFile Condition="'$(_PlatformName)' != 'macOS' And '$(InvariantGlobalization)' != 'true' And '$(HybridGlobalization)' == 'true' And '$(_GlobalizationDataFile)' == ''">icudt_hybrid.dat</_GlobalizationDataFile>
+        <_GlobalizationDataFile Condition="'$(_PlatformName)' != 'macOS' And '$(InvariantGlobalization)' != 'true' And '$(HybridGlobalization)' != 'true' And '$(_GlobalizationDataFile)' == '' And $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '9.0'))">icudt.dat</_GlobalizationDataFile>
+		<_GlobalizationDataFile Condition="'$(_PlatformName)' != 'macOS' And '$(InvariantGlobalization)' != 'true' And '$(HybridGlobalization)' == 'true' And '$(_GlobalizationDataFile)' == '' And $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '9.0'))">icudt_hybrid.dat</_GlobalizationDataFile>
```

Comment:
Instead of adding the condition on .NET 9, you can just target the `net9.0` branch.

That way I believe you can just remove these lines, and also any other lines related to the icu file:

```shell
$ git grep GlobalizationDataFile
dotnet/targets/Xamarin.Shared.Sdk.targets:              <_GlobalizationDataFile Condition="'$(_PlatformName)' != 'macOS' And '$(InvariantGlobalization)' != 'true' And '$(HybridGlobalization)' != 'true' And '$(_GlobalizationDataFile)' == ''">icudt.dat</_GlobalizationDataFile>
dotnet/targets/Xamarin.Shared.Sdk.targets:              <_GlobalizationDataFile Condition="'$(_PlatformName)' != 'macOS' And '$(InvariantGlobalization)' != 'true' And '$(HybridGlobalization)' == 'true' And '$(_GlobalizationDataFile)' == ''">icudt_hybrid.dat</_GlobalizationDataFile>
dotnet/targets/Xamarin.Shared.Sdk.targets:                      <_ArchitectureSpecificFiles Include="$(_GlobalizationDataFile)" Condition="'$(_GlobalizationDataFile)' != ''" />
dotnet/targets/Xamarin.Shared.Sdk.targets:              <Error Text="Can not set values for both InvariantGlobalization '$(InvariantGlobalization)' and _GlobalizationDataFile '$(_GlobalizationDataFile)'" Condition="'$(_GlobalizationDataFile)' != '' And '$(InvariantGlobalization)' == 'true'" />
dotnet/targets/Xamarin.Shared.Sdk.targets:                              GlobalizationDataFile=$(_GlobalizationDataFile)
dotnet/targets/Xamarin.Shared.Sdk.targets:                                          '%(ResolvedFileToPublish.Filename)%(ResolvedFileToPublish.Extension)' == '$(_GlobalizationDataFile)' And
msbuild/Xamarin.Shared/Xamarin.Shared.targets:                  <_GlobalizationDataFileAppBundleRelativePath Condition="'$(_AppContentsRelativePath)' == ''">$(_GlobalizationDataFile)</_GlobalizationDataFileAppBundleRelativePath>
msbuild/Xamarin.Shared/Xamarin.Shared.targets:                  <_GlobalizationDataFileAppBundleRelativePath Condition="'$(_AppContentsRelativePath)' != ''">$(_AppContentsRelativePath)\$(_GlobalizationDataFile)</_GlobalizationDataFileAppBundleRelativePath>
msbuild/Xamarin.Shared/Xamarin.Shared.targets:                  <RuntimeHostConfigurationOption Include="ICU_DAT_FILE_PATH" Value="$(_GlobalizationDataFileAppBundleRelativePath.Replace('\','/'))" />
tools/dotnet-linker/LinkerConfiguration.cs:             public string GlobalizationDataFile { get; private set; } = string.Empty;
tools/dotnet-linker/LinkerConfiguration.cs:                             case "GlobalizationDataFile":
tools/dotnet-linker/LinkerConfiguration.cs:                                     GlobalizationDataFile = value;
```

Looking at the [runtime PR](https://github.com/dotnet/runtime/pull/104443), it also looks like we'll have to stop linking with these libraries:


```shell
$ git grep -E 'icudata|icui18n|icuuc'
runtime/Makefile:DOTNET_iphonesimulator_DYLIB_FLAGS=-lmonosgen-2.0 -licudata -licui18n -licuuc -framework UIKit
runtime/Makefile:DOTNET_iphoneos_DYLIB_FLAGS=-lmonosgen-2.0 -licudata -licui18n -licuuc -framework UIKit
runtime/Makefile:DOTNET_tvsimulator_DYLIB_FLAGS=-lmonosgen-2.0 -licudata -licui18n -licuuc -framework UIKit
```

Reply:
@rolfbjarne by mistake I pushed https://github.com/xamarin/xamarin-macios/commit/66e7a1aeb214576b78d44b951501ec14e26944a0 to net9.0 instead of my branch, could you please help me revert that?

Reply:
I reverted that commit and pushed the revert.

Reply:
Thanks a lot!